MoviesStoreApp/MoviesStoreMVVM/CoreData/CoreDataService.swift

```diff
@@ -0,0 +1,187 @@
+// CoreDataService.swift
+// Copyright © Aleksandr Shchepelin. All rights reserved.
+
+import CoreData
+
+/// Сервис по обрабокте данных в CoreData
+final class CoreDataService: CoreDataServiceProtocol {
+    // MARK: - Public Properties
+
+    lazy var managedContext: NSManagedObjectContext = self.storeContainer.viewContext
+    var errorHandler: CoreDataHandler?
+
+    // MARK: - Private Properties
+
+    private let modelName: String
+    private lazy var storeContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: self.modelName)
+        container.loadPersistentStores { _, error in
+            if let error = error as NSError? {
+                self.errorHandler?(error.localizedDescription)
+            }
+        }
+        return container
+    }()
+
+    // MARK: - Init
+
+    init(modelName: String) {
+        self.modelName = modelName
+    }
+
+    // MARK: - Public Methods
+
+    func saveMovieDataContext(movies: [Movie], movieType: String) {
+        guard let newMovie = NSEntityDescription.entity(forEntityName: CoreDataConstants.movieData, in: managedContext)
+        else { return }
+        for movie in movies {
+            let movieContainer = MovieData(
+                entity: newMovie,
+                insertInto: managedContext
+            )
+            movieContainer.title = movie.title
+            movieContainer.overview = movie.overview
+            movieContainer.voteCount = Int64(movie.voteCount)
+            movieContainer.voteAverage = movie.voteAverage
+            movieContainer.movieId = Int64(movie.movieId)
+            movieContainer.poster = movie.poster
+            movieContainer.originalTitle = movie.originalTitle
+            movieContainer.adult = movie.adult
+            movieContainer.popularity = movie.popularity
+            movieContainer.releaseDate = movie.releaseDate
+            movieContainer.movieType = movieType
+            movieContainer.id = UUID()
+            do {
+                try managedContext.save()
+            } catch let error as NSError {
+                errorHandler?(error.localizedDescription)
+            }
+        }
+    }
+
+    func saveMovieInfoDataContext(movieInfo: MovieInfo, movieID: Int) {
```

Comment:
```suggestion
    func saveMovieInfoDataContext(movieInfo: MovieInfo) {
```
передаешь в метод но не используешь нигде, у тебя Айдишник Фильма уже в в МувиИнфо